bench/bench_htbl.ml

```diff
@@ -0,0 +1,76 @@
+open Multicore_bench
+
+let run_one ~budgetf ~n_domains ?(n_ops = 20 * Util.iter_factor)
+    ?(n_keys = 10000) ~percent_mem ?(percent_add = (100 - percent_mem + 1) / 2)
+    ?(prepopulate = true) (module Htbl : Htbl_intf.HTBL) =
+  let percent_rem = 100 - (percent_mem + percent_add) in
+
+  let limit_mem = percent_mem in
+  let limit_add = percent_mem + percent_add in
+
+  assert (0 <= limit_mem && limit_mem <= 100);
+  assert (limit_mem <= limit_add && limit_add <= 100);
+
+  let t = Htbl.create ~hashed_type:(module Int) () in
```

Comment:
We are already generating random integer values for the keys, which means that hashing the keys is unnecessary.  We also want to specifically measure the cost of the operations inside the hash table itself and exclude everything else as much as possible, which also includes `Int.hash` (which is also missing from older compiler versions).  So, it would be better to use something like:

```ocaml
module Key = struct
  type t = int

  let equal = Int.equal
  let hash = Fun.id
end
```

for the `hashed_type`.